src/Gemstone.Data/DataSetExtensions/DataSetExtensions.cs

```diff
@@ -172,335 +172,335 @@ public static class DataSetExtensions
-                                    output.Write(blob);
-                                }
-
-                                break;
-                        }
+                        case DataType.Boolean:
+                            output.Write(value.NotDBNull<bool>());
+
+                            break;
+                        case DataType.Byte:
+                            output.Write(value.NotDBNull<byte>());
+
+                            break;
+                        case DataType.Char:
+                            output.Write(value.NotDBNull<char>());
+
+                            break;
+                        case DataType.DateTime:
+                            output.Write(value.NotDBNull<DateTime>().Ticks);
+
+                            break;
+                        case DataType.Decimal:
+                            output.Write(value.NotDBNull<decimal>());
+
+                            break;
+                        case DataType.Double:
+                            output.Write(value.NotDBNull<double>());
+
+                            break;
+                        case DataType.Guid:
+                            output.Write(value.NotDBNull<Guid>().ToByteArray());
+
+                            break;
+                        case DataType.Int16:
+                            output.Write(value.NotDBNull<short>());
+
+                            break;
+                        case DataType.Int32:
+                            output.Write(value.NotDBNull<int>());
+
+                            break;
+                        case DataType.Int64:
+                            output.Write(value.NotDBNull<long>());
+
+                            break;
+                        case DataType.SByte:
+                            output.Write(value.NotDBNull<sbyte>());
+
+                            break;
+                        case DataType.Single:
+                            output.Write(value.NotDBNull<float>());
+
+                            break;
+                        case DataType.String:
+                            output.Write(value.NotDBNullString());
+
+                            break;
+                        case DataType.TimeSpan:
+                            output.Write(value.NotDBNull<TimeSpan>().Ticks);
+
+                            break;
+                        case DataType.UInt16:
+                            output.Write(value.NotDBNull<ushort>());
+
+                            break;
+                        case DataType.UInt32:
+                            output.Write(value.NotDBNull<uint>());
+
+                            break;
+                        case DataType.UInt64:
+                            output.Write(value.NotDBNull<ulong>());
+
+                            break;
+                        case DataType.Blob:
+                            byte[] blob = value.NotDBNull<byte[]>();
+
+                            if (blob.Length == 0)
```

Comment:
It's because the `NotDBNull` extension marks the return as non-nullable - so the compiler picks this up as not needed. Perhaps the better answer is to correct the signature of `NotDBNull` or to assume that not `DBNull` also means not `null`.

Thoughts?

Reply:
Since it was a local function - I marked it as nullable for now